pkg/cmd/util/clientcmd/factory.go

```diff
@@ -450,6 +468,19 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	return w
 }
 
+// overlyCautiousIllegalFileCharacters matches characters that *might* not be supported.  Windows is really restrictive, so this is really restrictive
+var overlyCautiousIllegalFileCharacters = regexp.MustCompile(`[^(\w/\.)]`)
```

Comment:
> not as cautious as you might hope...
> computeDiscoverCacheDir(".","foo/bar") == "foo/bar"

and?  Slash gets used as a path demiliter and you get a sane directory structure right?